### PR TITLE
knowledge: manager addition

### PR DESCRIPTION
**Pull request by jalavik**
* NEW Adds manager to knowledge with a command to load
  mappings into an existing knowledge base from a file.

  E.g. `inveniomanage knowledge load kb_name /path/to/file.kb`

Signed-off-by: Jan Aage Lavik <jan.age.lavik@cern.ch>

### CODE REVIEW

File: invenio/modules/knowledge/manage.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        print("KB does not exist\n",
+              file=sys.stderr)
+        sys.exit(1)
+    num_added = add_kb_values(name, filepath, separator)
+    print(">>> Knowledge '{0}' updated successfully with {1} entries.".format(
+        name, num_added
+    ))
+
+
+def add_kb_values(kbname, kbfile, separator):
```
Comment (jirikuncar) [inline invenio/modules/knowledge/manage.py]: do we have any utility function in `.utils` or can we share any loader function from admin interface? 

--
cc @hachreak

Comment (hachreak) [inline invenio/modules/knowledge/manage.py]: didn't exists a `loader` function.
Maybe we can create `.utils` and move there.

Comment (jalavik) [inline invenio/modules/knowledge/manage.py]: Sure, I could add it to `knowledge.utils` as something like `load_kb_mappings_file`.

File: invenio/modules/knowledge/manage.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+from sqlalchemy.orm.exc import NoResultFound
+
+from .api import add_kb_mapping, get_kb_by_name
+
+manager = Manager(usage=__doc__)
+
+
+@manager.command
+@manager.option('--name', '-w', dest='name')
```
Comment (PXke) [inline invenio/modules/knowledge/manage.py]: Surely a stupid question but why -w and -t  for name and separator ? :wink:

Comment (jalavik) [inline invenio/modules/knowledge/manage.py]: Whoops, no you are correct. Bad names. I will rename to for example, `-n` and `-s`

**Comment by jalavik** [comment]
Updated PR.

**Comment by jirikuncar** [comment]
> 6: M102 unrecognized bullet/signature

@jalavik Can you remove the empty line (no. 5)? Thanks

File: invenio/modules/knowledge/utils.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Knowledge utilities."""
+
+from .api import add_kb_mapping
```
Comment (jirikuncar) [inline invenio/modules/knowledge/utils.py]: missing `from __future__ import print_function`

File: invenio/modules/knowledge/utils.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    num_added = 0
+    with open(kbfile) as kb_fd:
+        for line in kb_fd:
+            if not line.strip():
+                continue
+            try:
+                key, value = line.split(separator)
+            except ValueError:
+                # bad split, pass
+                print("Error splitting: {0}".format(line))
```
Comment (jirikuncar) [inline invenio/modules/knowledge/utils.py]: https://github.com/inveniosoftware/invenio/pull/2878/files#r26742392 or use `current_app.logger.error/exception`

Comment (jalavik) [inline invenio/modules/knowledge/utils.py]: Good point. I will use the `current_app.logger` everywhere. Thanks!

**Comment by jalavik** [comment]
Updated.

**Pull request merged**